api/v1/verticadb_webhook.go

```diff
@@ -1292,6 +1292,22 @@ func (v *VerticaDB) validateProxyLogLevel(allErrs field.ErrorList) field.ErrorLi
 	return allErrs
 }
 
+// check if sandbox has a primary subcluster
+func (sand *Sandbox) hasSandboxPrimarySubcluster(allErrs field.ErrorList) field.ErrorList {
+	for _, sc := range sand.Subclusters {
+		if sc.Type == PrimarySubcluster {
+			return allErrs
+		}
+	}
+
+	err := field.Invalid(field.NewPath("spec").Child("sandboxes"),
+		sand.Subclusters,
+		fmt.Sprintf("there must be at least one primary subcluster in the sandbox %s", sand.Name))
+	allErrs = append(allErrs, err)
+
+	return allErrs
+}
```

Comment:
Refactor this to a boolean func that returns true if the sandbox has at least one primary sc. 
```suggestion
func (sand *Sandbox) hasPrimarySubcluster() bool {
	for i := range sand.Subclusters {
	        sc := &sand.Subclusters[i]
		if sc.Type == PrimarySubcluster {
			return true
		}
	}

	return false
}

func (v *VerticaDB) validateSandboxPrimarySubcluster(allErrs field.ErrorList, sbIndex int) field.ErrorList {
	sb := &v.Spec.Sandboxes[i]
	if sb != nil && !sb.hasPrimarySubcluster() {
		err := field.Invalid(field.NewPath("spec").Child("sandboxes").Index(sbIndex),
			sb.Subclusters,
			fmt.Sprintf("there must be at least one primary subcluster in the sandbox %s", sb.Name))
		allErrs = append(allErrs, err)
	}
	return allErrs
}
```
then call validateSandboxPrimarySubcluster in validateSubclustersInSandboxes

Reply:
Thanks. Refactored.